src/rpc/blockchain.cpp

```diff
@@ -1774,7 +1750,7 @@ static UniValue getblockstats(const JSONRPCRequest& request)
 
     UniValue ret_all(UniValue::VOBJ);
     ret_all.pushKV("avgfee", (block.vtx.size() > 1) ? totalfee / (block.vtx.size() - 1) : 0);
-    ret_all.pushKV("avgfeerate", total_weight ? (totalfee * WITNESS_SCALE_FACTOR) / total_weight : 0); // Unit: sat/vbyte
+    ret_all.pushKV("avgfeerate", total_size > 0 ? totalfee / total_size : 0); // Unit: tap/byte
```

Comment:
Are there any cases where total_size is less than or equal `0`?

Reply:
Yes,coinbase transactions sizes are 0 because of lines 1686:1688
`        if (tx->IsCoinBase()) {
            continue;
        }`